Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py

```diff
@@ -2889,49 +2913,114 @@ def download_fetched_file(client: Client, args: dict) -> List[CommandResults]:
             rank = 0
         # If no fetch threat rank is provided, bring everything, else only fetch above the threshold
-        if IS_VERSION_2_1 or rank >= fetch_threat_rank:
-            incident = threat_to_incident(threat)
+        if IS_VERSION_2_1 or rank >= args.get('fetch_threat_rank'):
+            incident = to_incident('Threat', threat)
             date_occurred_dt = parse(incident['occurred'])
             incident_date = int(date_occurred_dt.timestamp() * 1000)
-            if incident_date > last_fetch:
-                incidents.append(incident)
+            if incident_date > args.get('last_fetch'):
+                incidents_threats.append(incident)
 
             if incident_date > current_fetch:
                 current_fetch = incident_date
 
+    return incidents_threats, current_fetch
+
+
+def fetch_alerts(client: Client, args: dict):
+
+    incidents_alerts = []
+    current_fetch = args.get('current_fetch')
+
+    query_params = assign_params(
+        incidentStatus=','.join(args.get('fetch_incidentStatus')),
+        createdAt__gte=args.get('last_fetch_date_string'),
+        limit=args.get('fetch_limit'),
+        siteIds=args.get('fetch_site_ids')
+    )
+
+    alerts, pagination = client.get_alerts_request(query_params)
+    for alert in alerts:
+        severity = alert.get('ruleInfo').get('severity')
+
+        if severity in args.get('fetch_severity'):
+            incident = to_incident('Alert', alert)
+            date_occurred_dt = parse(incident['occurred'])
+            incident_date = int(date_occurred_dt.timestamp() * 1000)
+            if incident_date > args.get('last_fetch'):
+                incidents_alerts.append(incident)
+
+            if incident_date > current_fetch:
+                current_fetch = incident_date
+
+    return incidents_alerts, current_fetch
+
+
+def fetch_handler(client: Client, args: dict):
+
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('time')
+
+    if last_fetch is None:
+        last_fetch = dateparser.parse(args.get('first_fetch_time'), settings={'TIMEZONE': 'UTC'})
+        if not last_fetch:
+            raise DemistoException('Please provide an initial First fetch timestamp')
+        last_fetch = int(last_fetch.timestamp() * 1000)
+
+    current_fetch = last_fetch
+    last_fetch_date_string = timestamp_to_datestring(last_fetch, '%Y-%m-%dT%H:%M:%S.%fZ')
+
+    args['last_fetch'] = last_fetch
+    args['last_fetch_date_string'] = last_fetch_date_string
+    args['current_fetch'] = current_fetch
+
+    if len(args.get('fetch_type')) == 1:
+        if 'Alerts' in args.get('fetch_type'):
+            incidents, current_fetch = fetch_alerts(client, args)
+        elif 'Threats' in args.get('fetch_type'):
+            incidents, current_fetch = fetch_threats(client, args)
+    else:
+        alert_incidents, alert_current_fetch = fetch_alerts(client, args)
+        threat_incidents, threat_current_fetch = fetch_threats(client, args)
```

Comment:
change this part, it should be something like:
```
if fetch_type == 'Both': 
    alert_incidents, alert_current_fetch = fetch_alerts(client, args)
    threat_incidents, threat_current_fetch = fetch_threats(client, args)
elif fetch_type == 'Alerts': 
    incidents, current_fetch = fetch_alerts(client, args)
elif fetch_type == 'Threats':
    incidents, current_fetch = fetch_threats(client, args)
```

Reply:
@mohlcyber what about this part?

Reply:
@adi88d I did the change in the code. Appreciate your support.